test/dotnet-publish.Tests/GivenDotnetPublishPublishesProjects.cs

```diff
@@ -85,6 +83,65 @@ public void ItPublishesARunnableSelfContainedApp()
                      .And.HaveStdOutContaining("Hello World");
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ItPublishesAnAppExplicitlySpecifyingSelfContained(bool selfContained)
+        {
+            var testAppName = "MSBuildTestApp";
+
+            var testInstance = TestAssets.Get(testAppName)
+                .CreateInstance($"PublishesSelfContained{selfContained}")
+                .WithSourceFiles()
+                .WithRestoreFiles();
+
+            var testProjectDirectory = testInstance.Root;
+
+            var rid = DotnetLegacyRuntimeIdentifiers.InferLegacyRestoreRuntimeIdentifier();
+
+            new PublishCommand()
+                .WithRuntime(rid)
+                .WithSelfContained(selfContained)
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute()
+                .Should().Pass();
+
+            var configuration = Environment.GetEnvironmentVariable("CONFIGURATION") ?? "Debug";
+            var outputPath = testProjectDirectory
+                    .GetDirectory("bin", configuration, "netcoreapp2.0", rid, "publish")
+                    .FullName;
+            var selfContainedProgram = Path.Combine(outputPath, $"{testAppName}{Constants.ExeSuffix}");
+            var selfContainedProgramFile = new FileInfo(selfContainedProgram);
+
+            TestCommand testCommand;
+            string testArgs;
+            if (selfContained)
```

Comment:
You should break this test into two instead of do this if/else here. You can then extract the common piece of code above into a helper method and invoke it there.